src/Security/Authorization/Core/src/AuthorizationPolicy.cs

```diff
@@ -176,7 +176,17 @@ public static async Task<AuthorizationPolicy> CombineAsync(IAuthorizationPolicyP
                 }
             }
 
+            // If we have no policy by now, use the fallback policy if we have one
```

Comment:
Do we know the set of places where this method is called?
1. MVC (AuthorizeFilter)
2. AuthorizationMiddleware
3. ????

This change won't impact MVC because we never call this method with empty authorize data.

Reply:
Oh, I thought the middleware would call GetFallbackPolicyAsync directly?
What does SignalR do?

Reply:
SignalR calls this here - https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/common/Http.Connections/src/Internal/AuthorizeHelper.cs#L24, but it's the same behavior where it no-ops when no `AuthorizeData` is present. 

If the intent for this is to work specifically for AuthZMiddleware, 
1) Have we considered an option where this is an option passed to the `AuthZ` middleware? `app.AddAuthorizationPolicyEvaluator(option => option.FallbackPolicy = ...)`. It does make this a non-breaking change 

2) Alternatively, have the `AuthZMiddleware` call `GetFallbackPolicyAsync` when there's no data. This makes it so that no existing consumers of this API are inadvertently affected this change.

Both options mean that the feature applies specifically to `AuthZMiddleware` as documented in the remarks section of this feature.

Reply:
Perhaps I should just reword the comment to also explain that this policy is applied as part of AuthorizationPolicy.Combine.  Neither default nor fallback policy really makes sense when talking about the explicit `IAuthorizationService` since that takes either a policy name, or an explicit set of requirements.

 I don't think the intent for this feature is to 'only' apply to the middleware, if we wanted, I think we could if we wanted, also provide a parameterless overload that didn't specify a policy name/requirements `AuthorizeAsync(user, resource)` that would use the Fallback policy.

Reply:
I'm not proposing we change anything, but I want to make the following clear:

> I don't think the intent for this feature is to 'only' apply to the middleware, if we wanted

My understanding is that this feature **CANNOT** work in MVC using `IRouter`. The solution for that user is to apply the `AuthorizeFilter` globally. 

I think that this is a OK resolution.